x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/session/Configuration.java

```diff
@@ -76,6 +78,7 @@ public Configuration(
         this.profile = profile;
         this.tables = tables;
         assert tables != null;
+        this.queryStartTimeMillis = queryStartTimeMillis;
```

Comment:
Hmm, not sure. If you feel that is safe and future proof. Right now this class (in non-test code) is only created in one place (TransportEsqlQueryAction.innerExecute), so your suggestion would certainly work, but burying it in Configuration feels like a hidden piece of state that should be clear to the Action doing it. On the _search side, for example, there is an explicit [SearchTimeProvider](https://github.com/elastic/elasticsearch/blob/main/server/src/main/java/org/elasticsearch/action/search/TransportSearchAction.java#L294) that gets passed around to make it explicit.

So my vote is that I prefer it to be created in the Action class and passed in. That makes it clear that Configuration is used for tracking time and could allow testing it be easier if we need to pass in stubbed in time values.

Side note: the Configuration class already has an `absoluteStartedTimeInMillis` that returns `System.currentTimeMillis()` each time it is called, which seems like incorrect/unexpected behavior to me. That's a case where I think the Configuration class should not be creating that time (since it doesn't retain it).

Reply:
> Side note: the Configuration class already has an absoluteStartedTimeInMillis that returns System.currentTimeMillis() each time it is called, which seems like incorrect/unexpected behavior to me. That's a case where I think the Configuration class should not be creating that time (since it doesn't retain it).

Indeed, that doesn't look right.